project-fob/Controllers/AttendeeController.cs

```diff
@@ -26,21 +26,43 @@ public ActionResult Index()
 
         public void Fob(string value)
         {
-            HttpContext.Session.TryGetValue("meetingid", out var meetingIdValue);
+            if (HaveCookieId())
+            {
+                string userId = GetCookieId();
+                HttpContext.Session.TryGetValue("meetingid", out var meetingIdValue);
 
-            string meetingString = System.Text.Encoding.ASCII.GetString(meetingIdValue);
+                string meetingString = System.Text.Encoding.ASCII.GetString(meetingIdValue);
 
-            Fob fob = db.Fob.Include(x => x.Meeting).ThenInclude(x => x.Stats).Include(x => x.Fobbed).ThenInclude(x => x.User).Single(f => f.Meeting.MeetingId == meetingString);
+                Fob fob = db.Fob.Include(x => x.Meeting).ThenInclude(x => x.Stats).Single(f => f.Meeting.MeetingId == meetingString);
+                
+                fob.AddFob(userId);
+                db.SaveChanges();
 
-            HttpContext.Session.TryGetValue("sessionid", out var session);
+            }
+            else
+            {
+                Index();
+            }
+        }
 
-            string userIdString = System.Text.Encoding.ASCII.GetString(session);
-            Attendee att = db.Attendee.Include(at => at.User).Include(at => at.Meeting).Single(at => at.User.UserId.Equals(userIdString) && at.Meeting.MeetingId.Equals(meetingString));
+        private bool HaveCookieId()
+        {
+            string id = Request.Cookies["ID"];
+            if (id == null || id.Length == 0)
+            {
+                return false;
+            }
+            return true;
+        }
 
-            fob.AddFob(att);
-            db.SaveChanges();
+        private string GetCookieId()
+        {
+            //This method does not check if the cookie is correct or not. Please use HaveCookieId
+            return Request.Cookies["ID"];
```

Comment:
The `"ID"` lookup key is used in a few places, this would be better extracted out as a constant.